core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly

```diff
@@ -37,14 +37,16 @@ THE SOFTWARE.
           ${%Connect agent to Jenkins one of these ways:}
         </p>
         <ul>
-          <li>
-            <p>
-              <a href="slave-agent.jnlp" id="jnlp-link">
-                <img src="${imagesURL}/webstart.gif" alt="${%launch agent}" />
-              </a>
-              ${%Launch agent from browser}
-            </p>
-          </li>
+          <j:if test="${it.launcher.javaWebStartSupported}">
+            <li>
+              <p>
+                <a href="slave-agent.jnlp" id="jnlp-link">
+                  <img src="${imagesURL}/webstart.gif" alt="${%launch agent}" />
+                </a>
+                ${%Launch agent from browser}
+              </p>
+            </li>
+          </j:if>
```

Comment:
`else` add some warning to the WebUI? Otherwise we may end up with users who report the issue after reading the docs and failing to fond the button